internal/expr/bindtypes.go

```diff
@@ -143,17 +145,28 @@ func bindInputTypes(ti typeNameToInfo, e *inputExpr) (tm typeinfo.Member, err er
 		return nil, typeMissingError(e.sourceType.typeName, getKeys(ti))
 	}
 
-	tm, err = info.TypeMember(e.sourceType.memberName)
+	tm, err := info.TypeMember(e.sourceType.memberName)
 	if err != nil {
 		return nil, err
 	}
-	return tm, nil
+	return &typedInputExpr{tm}, nil
 }
 
-// bindOutputTypes binds the output expression to concrete types. It then checks
-// the expression is formatted correctly and generates the columns for the query
-// and the typeMembers the columns correspond to.
-func bindOutputTypes(ti typeNameToInfo, e *outputExpr) (outCols []columnAccessor, typeMembers []typeinfo.Member, err error) {
+// outputExpr represents a named target output variable in the SQL expression,
+// as well as the source table and column where it will be read from.
+type outputExpr struct {
+	sourceColumns []columnAccessor
+	targetTypes   []valueAccessor
+	raw           string
+}
+
+func (e *outputExpr) String() string {
+	return fmt.Sprintf("Output[%+v %+v]", e.sourceColumns, e.targetTypes)
+}
+
+// bindTypes binds the output expression to concrete types. It then checks the
+// expression is formatted correctly and generates a typed output expression.
+func (e *outputExpr) bindTypes(ti typeNameToInfo) (te typedExpression, err error) {
```

Comment:
Can we change the return of the function to `*typedOutputExpr` which makes it clearer that it will only return that implementor of the interface?

Reply:
Go doesn't allow you to do this and still implement the interface.

Reply:
Sad, thanks